crates/uv/src/commands/project/remove.rs

```diff
@@ -93,29 +95,30 @@ pub(crate) async fn remove(
     }?;
 
     for package in packages {
+        let requirement = Requirement::from_str(package.as_str())?;
```

Comment:
I don't think we should do this cast / change the removal APIs, why was this needed? Removal operates on package names, not full requirements.

Reply:
I see this probably is related to https://github.com/astral-sh/uv/pull/6005/commits/0fbde82c7dbe1ac9fd40919d642f5976d2b79d08 now, perhaps we can cast to a `Requirement` later instead if needed?